indexing-hadoop/src/main/java/io/druid/indexer/updater/HadoopConverterJob.java

```diff
@@ -481,13 +476,13 @@ protected void map(
       final String tmpDirLoc = context.getConfiguration().get(TMP_FILE_LOC_KEY);
       final File tmpDir = Paths.get(tmpDirLoc).toFile();
 
-      final DataSegment segment = ((DataSegmentSplit) split).getDataSegment();
+      final DataSegment segment = ((DatasourceInputSplit) split).getSegments().get(0);
```

Comment:
is it always a single element here  ? 
probably use Iterables.getOnlyElement() to make sure we always have single segment here.